src/common/dump.c

```diff
@@ -122,18 +123,14 @@ unsigned int dump_in_xflash_read_regs_GEN(void *pRegsGEN, unsigned int size) {
 }
 
 void dump_in_xflash_reset(void) {
-    if (w25x_init()) {
-        for (uint32_t addr = 0; addr < DUMP_XFLASH_SIZE; addr += 0x10000) {
-            w25x_block64_erase(DUMP_OFFSET + addr);
-        }
+    for (uint32_t addr = 0; addr < DUMP_XFLASH_SIZE; addr += 0x10000) {
```

Comment:
Is it safe to remove the `w25x_init()` from multiple `dump_*` functions? If they are "private" and not part of the dump API, it is ok. However, if they can be called separately in arbitrary order, relying on having the w25x initialized may be dangerous.

Reply:
w25x_init() is called in the beginning of main() just after initializing integrated peripherals. I think it is OK to rely on the fact those dump functions will not be called before the basic system is initialized.
Reasoning was already explained in commit messages:

[Remove redundant.](https://github.com/prusa3d/Prusa-Firmware-Buddy/pull/2042/commits/6cc1538f797efaec7fa6a84274bcf13f27981074)
It is asured in main(), that w25x is initilized succesfully
otherwise BSOD is called and we don't reach this point.

[Remove redundant.](https://github.com/prusa3d/Prusa-Firmware-Buddy/pull/2042/commits/4929461b8c85510837fb167eb55660a255ee450e)
There is no need to assure w25x_init() as
dump_save_to_usb() is called from GUI
and it is assured w25x to be successfully initialized
before GUI is started.


[Remove redundant.](https://github.com/prusa3d/Prusa-Firmware-Buddy/pull/2042/commits/d8c6170baa89deec07a75b73a42fbea1b8049e9f)
There is no need to assure w25x_init() as
dump_in_xflash_reset() is called from main only after w25x
was successfully initialized.
It is also called from GUI and it is assured w25x being
initialized before GUI is started.

[Remove redundant.](https://github.com/prusa3d/Prusa-Firmware-Buddy/pull/2042/commits/424464098347d99f5f4273dd40138aebdb3b5e40)
There is no need to assure w25x_init() as
dump_in_xflash_delete() is called from GUI
and it is assured w25x to be successfully initialized
before GUI is started.